From beec7a7aaa05e945fdb40954336408663feafcdd Mon Sep 17 00:00:00 2001
From: Oleg Davydov <burunduk@chromium.org>
Date: Mon, 7 Jun 2021 16:15:43 +0000
Subject: [PATCH] Migrate policy tools to python3

This CL converts the rest (after https://crrev.com/c/2940544) scripts
from components/policy/tools/ from python2 to python3.

Bug: 1196322
Change-Id: I1238583ac9471e88120097fe64a27f046118fef1
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2940552
Reviewed-by: Yann Dago <ydago@chromium.org>
Commit-Queue: Oleg Davydov <burunduk@chromium.org>
Cr-Commit-Position: refs/heads/master@{#889781}
---
 .../tools/template_writers/PRESUBMIT.py       |  5 +++-
 .../policy/tools/template_writers/__init__.py |  2 +-
 .../policy_template_generator.py              |  2 +-
 .../policy_template_generator_unittest.py     |  2 +-
 .../template_writers/template_formatter.py    |  2 +-
 .../tools/template_writers/test_suite_all.py  |  2 +-
 .../template_writers/writer_configuration.py  |  2 +-
 .../template_writers/writers/__init__.py      |  2 +-
 .../template_writers/writers/adm_writer.py    |  2 +-
 .../writers/adm_writer_unittest.py            |  2 +-
 .../template_writers/writers/adml_writer.py   |  2 +-
 .../writers/adml_writer_unittest.py           |  2 +-
 .../template_writers/writers/admx_writer.py   |  2 +-
 .../writers/admx_writer_unittest.py           |  2 +-
 .../writers/android_policy_writer.py          |  2 +-
 .../writers/android_policy_writer_unittest.py |  2 +-
 .../writers/chromeos_adml_writer.py           |  2 +-
 .../writers/chromeos_adml_writer_unittest.py  |  2 +-
 .../writers/chromeos_admx_writer.py           |  2 +-
 .../writers/chromeos_admx_writer_unittest.py  |  2 +-
 .../writers/doc_atomic_groups_writer.py       |  2 +-
 .../template_writers/writers/doc_writer.py    | 23 ++++++++++++++-----
 .../writers/doc_writer_unittest.py            |  2 +-
 .../writers/google_adml_writer.py             |  2 +-
 .../writers/google_adml_writer_unittest.py    |  2 +-
 .../writers/google_admx_writer.py             |  2 +-
 .../writers/google_admx_writer_unittest.py    |  2 +-
 .../writers/gpo_editor_writer.py              |  2 +-
 .../writers/ios_app_config_writer.py          |  2 +-
 .../writers/ios_app_config_writer_unittest.py |  2 +-
 .../template_writers/writers/jamf_writer.py   |  5 ++--
 .../writers/jamf_writer_unittest.py           |  2 +-
 .../template_writers/writers/json_writer.py   |  2 +-
 .../writers/json_writer_unittest.py           |  2 +-
 .../template_writers/writers/mock_writer.py   |  4 ++--
 .../template_writers/writers/plist_helper.py  |  2 +-
 .../writers/plist_strings_writer.py           |  2 +-
 .../template_writers/writers/plist_writer.py  |  2 +-
 .../writers/plist_writer_unittest.py          |  2 +-
 .../template_writers/writers/reg_writer.py    |  2 +-
 .../writers/reg_writer_unittest.py            |  2 +-
 .../writers/template_writer.py                |  8 +++----
 .../writers/template_writer_unittest.py       |  2 +-
 .../writers/writer_unittest_common.py         |  2 +-
 .../writers/xml_formatted_writer.py           |  2 +-
 .../writers/xml_writer_base_unittest.py       |  2 +-
 46 files changed, 71 insertions(+), 56 deletions(-)

diff --git a/src/components/policy/tools/template_writers/PRESUBMIT.py b/src/components/policy/tools/template_writers/PRESUBMIT.py
index 66f6f827f3..e159b0d3b7 100755
--- a/src/components/policy/tools/template_writers/PRESUBMIT.py
+++ b/src/components/policy/tools/template_writers/PRESUBMIT.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -9,6 +9,9 @@ details on the presubmit API built into gcl.
 """
 
 
+USE_PYTHON3 = True
+
+
 def RunUnittests(input_api, output_api):
   return input_api.canned_checks.RunPythonUnitTests(input_api, output_api,
                                                     ['test_suite_all'])
diff --git a/src/components/policy/tools/template_writers/__init__.py b/src/components/policy/tools/template_writers/__init__.py
index 14d138a31c..abc93655ae 100755
--- a/src/components/policy/tools/template_writers/__init__.py
+++ b/src/components/policy/tools/template_writers/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/policy_template_generator.py b/src/components/policy/tools/template_writers/policy_template_generator.py
index eb2cba04bf..8f955f799d 100755
--- a/src/components/policy/tools/template_writers/policy_template_generator.py
+++ b/src/components/policy/tools/template_writers/policy_template_generator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/policy_template_generator_unittest.py b/src/components/policy/tools/template_writers/policy_template_generator_unittest.py
index 09d9cc58e3..5beb6082a4 100755
--- a/src/components/policy/tools/template_writers/policy_template_generator_unittest.py
+++ b/src/components/policy/tools/template_writers/policy_template_generator_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/template_formatter.py b/src/components/policy/tools/template_writers/template_formatter.py
index 56b63c5fd7..fe1f7a5606 100755
--- a/src/components/policy/tools/template_writers/template_formatter.py
+++ b/src/components/policy/tools/template_writers/template_formatter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/test_suite_all.py b/src/components/policy/tools/template_writers/test_suite_all.py
index 87b2317e7f..8ca43de457 100755
--- a/src/components/policy/tools/template_writers/test_suite_all.py
+++ b/src/components/policy/tools/template_writers/test_suite_all.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writer_configuration.py b/src/components/policy/tools/template_writers/writer_configuration.py
index b3917d1c69..a2c07c4ca5 100755
--- a/src/components/policy/tools/template_writers/writer_configuration.py
+++ b/src/components/policy/tools/template_writers/writer_configuration.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/__init__.py b/src/components/policy/tools/template_writers/writers/__init__.py
index df56f5caa8..fb050d1f24 100755
--- a/src/components/policy/tools/template_writers/writers/__init__.py
+++ b/src/components/policy/tools/template_writers/writers/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/adm_writer.py b/src/components/policy/tools/template_writers/writers/adm_writer.py
index 4b480c4587..7c20a5a8e7 100755
--- a/src/components/policy/tools/template_writers/writers/adm_writer.py
+++ b/src/components/policy/tools/template_writers/writers/adm_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/adm_writer_unittest.py b/src/components/policy/tools/template_writers/writers/adm_writer_unittest.py
index 38389e0d4e..9d2b86d9e3 100755
--- a/src/components/policy/tools/template_writers/writers/adm_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/adm_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/adml_writer.py b/src/components/policy/tools/template_writers/writers/adml_writer.py
index 96af4b6b07..e40dcb116e 100755
--- a/src/components/policy/tools/template_writers/writers/adml_writer.py
+++ b/src/components/policy/tools/template_writers/writers/adml_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/adml_writer_unittest.py b/src/components/policy/tools/template_writers/writers/adml_writer_unittest.py
index 6a34cccc3b..2f1afd1450 100755
--- a/src/components/policy/tools/template_writers/writers/adml_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/adml_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/admx_writer.py b/src/components/policy/tools/template_writers/writers/admx_writer.py
index 8da5024ca0..58b761c869 100755
--- a/src/components/policy/tools/template_writers/writers/admx_writer.py
+++ b/src/components/policy/tools/template_writers/writers/admx_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/admx_writer_unittest.py b/src/components/policy/tools/template_writers/writers/admx_writer_unittest.py
index 0ba159d4d5..1543a016bb 100755
--- a/src/components/policy/tools/template_writers/writers/admx_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/admx_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/android_policy_writer.py b/src/components/policy/tools/template_writers/writers/android_policy_writer.py
index 593e04c6ef..fa9ee07e62 100755
--- a/src/components/policy/tools/template_writers/writers/android_policy_writer.py
+++ b/src/components/policy/tools/template_writers/writers/android_policy_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2015 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/android_policy_writer_unittest.py b/src/components/policy/tools/template_writers/writers/android_policy_writer_unittest.py
index b36a72a846..a282bc5aab 100755
--- a/src/components/policy/tools/template_writers/writers/android_policy_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/android_policy_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2015 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/chromeos_adml_writer.py b/src/components/policy/tools/template_writers/writers/chromeos_adml_writer.py
index 6ae5309082..72e4dfc583 100755
--- a/src/components/policy/tools/template_writers/writers/chromeos_adml_writer.py
+++ b/src/components/policy/tools/template_writers/writers/chromeos_adml_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/chromeos_adml_writer_unittest.py b/src/components/policy/tools/template_writers/writers/chromeos_adml_writer_unittest.py
index 4cf97d3d0f..f07d299270 100755
--- a/src/components/policy/tools/template_writers/writers/chromeos_adml_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/chromeos_adml_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/chromeos_admx_writer.py b/src/components/policy/tools/template_writers/writers/chromeos_admx_writer.py
index fa53e170d3..c9f1c2c505 100755
--- a/src/components/policy/tools/template_writers/writers/chromeos_admx_writer.py
+++ b/src/components/policy/tools/template_writers/writers/chromeos_admx_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/chromeos_admx_writer_unittest.py b/src/components/policy/tools/template_writers/writers/chromeos_admx_writer_unittest.py
index 6208f02914..b6b4f83a42 100755
--- a/src/components/policy/tools/template_writers/writers/chromeos_admx_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/chromeos_admx_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/doc_atomic_groups_writer.py b/src/components/policy/tools/template_writers/writers/doc_atomic_groups_writer.py
index a98f1a6a16..9f92347a6c 100755
--- a/src/components/policy/tools/template_writers/writers/doc_atomic_groups_writer.py
+++ b/src/components/policy/tools/template_writers/writers/doc_atomic_groups_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2019 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/doc_writer.py b/src/components/policy/tools/template_writers/writers/doc_writer.py
index 5f7d775df0..178497ca80 100755
--- a/src/components/policy/tools/template_writers/writers/doc_writer.py
+++ b/src/components/policy/tools/template_writers/writers/doc_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2019 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -127,7 +127,11 @@ class DocWriter(xml_formatted_writer.XMLFormattedWriter):
     '''
     dd = self._AddPolicyAttribute(parent, 'schema', None,
                                   ['.monospace', '.pre-wrap'])
-    schema_json = json.dumps(schema, indent=2, sort_keys=True)
+    # Explicitly specify separators since defaults depend on python version.
+    schema_json = json.dumps(schema,
+                             indent=2,
+                             sort_keys=True,
+                             separators=(", ", ": "))
     self.AddText(dd, schema_json)
 
   def _AddFeatures(self, parent, policy):
@@ -141,8 +145,7 @@ class DocWriter(xml_formatted_writer.XMLFormattedWriter):
     '''
     features = []
     # The sorting is to make the order well-defined for testing.
-    keys = policy['features'].keys()
-    keys.sort()
+    keys = sorted(policy['features'].keys())
     for key in keys:
       key_name = self._FEATURE_MAP[key]
       if policy['features'][key]:
@@ -325,7 +328,11 @@ class DocWriter(xml_formatted_writer.XMLFormattedWriter):
     self.AddElement(parent, 'dt', {}, os_header)
     element = self._AddStyledElement(parent, 'dd', ['.monospace', '.pre-wrap'])
     key_name = self._GetRegistryKeyName(policy, is_win)
-    example = json.dumps(policy['example_value'], indent=2, sort_keys=True)
+    # Explicitly specify separators since defaults depend on python version.
+    example = json.dumps(policy['example_value'],
+                         indent=2,
+                         sort_keys=True,
+                         separators=(", ", ": "))
     self.AddText(element, '%s\\%s = %s' % (key_name, policy['name'], example))
 
   def _AddDictionaryExampleAndroidLinux(self, parent, policy):
@@ -339,7 +346,11 @@ class DocWriter(xml_formatted_writer.XMLFormattedWriter):
     '''
     self.AddElement(parent, 'dt', {}, 'Android/Linux:')
     element = self._AddStyledElement(parent, 'dd', ['.monospace', '.pre-wrap'])
-    example = json.dumps(policy['example_value'], indent=2, sort_keys=True)
+    # Explicitly specify separators since defaults depend on python version.
+    example = json.dumps(policy['example_value'],
+                         indent=2,
+                         sort_keys=True,
+                         separators=(", ", ": "))
     self.AddText(element, '%s: %s' % (policy['name'], example))
 
   def _AddDictionaryExample(self, parent, policy):
diff --git a/src/components/policy/tools/template_writers/writers/doc_writer_unittest.py b/src/components/policy/tools/template_writers/writers/doc_writer_unittest.py
index ed8de5ccdb..e6d01728e7 100755
--- a/src/components/policy/tools/template_writers/writers/doc_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/doc_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/google_adml_writer.py b/src/components/policy/tools/template_writers/writers/google_adml_writer.py
index 85d1f960d1..a126d34096 100755
--- a/src/components/policy/tools/template_writers/writers/google_adml_writer.py
+++ b/src/components/policy/tools/template_writers/writers/google_adml_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/google_adml_writer_unittest.py b/src/components/policy/tools/template_writers/writers/google_adml_writer_unittest.py
index 8ca5515573..3170f57f5f 100755
--- a/src/components/policy/tools/template_writers/writers/google_adml_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/google_adml_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/google_admx_writer.py b/src/components/policy/tools/template_writers/writers/google_admx_writer.py
index 4e013a80c2..e1f885ff60 100755
--- a/src/components/policy/tools/template_writers/writers/google_admx_writer.py
+++ b/src/components/policy/tools/template_writers/writers/google_admx_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/google_admx_writer_unittest.py b/src/components/policy/tools/template_writers/writers/google_admx_writer_unittest.py
index f82e24c04f..f4f1e6dcdc 100755
--- a/src/components/policy/tools/template_writers/writers/google_admx_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/google_admx_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/gpo_editor_writer.py b/src/components/policy/tools/template_writers/writers/gpo_editor_writer.py
index b662eb02ca..7ace447c3f 100755
--- a/src/components/policy/tools/template_writers/writers/gpo_editor_writer.py
+++ b/src/components/policy/tools/template_writers/writers/gpo_editor_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2018 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/ios_app_config_writer.py b/src/components/policy/tools/template_writers/writers/ios_app_config_writer.py
index 233d55757f..a5a61f9d40 100755
--- a/src/components/policy/tools/template_writers/writers/ios_app_config_writer.py
+++ b/src/components/policy/tools/template_writers/writers/ios_app_config_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2020 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/ios_app_config_writer_unittest.py b/src/components/policy/tools/template_writers/writers/ios_app_config_writer_unittest.py
index aadaf534d1..8774f2e84a 100755
--- a/src/components/policy/tools/template_writers/writers/ios_app_config_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/ios_app_config_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2020 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/jamf_writer.py b/src/components/policy/tools/template_writers/writers/jamf_writer.py
index e87cd15c6f..41fa458dec 100755
--- a/src/components/policy/tools/template_writers/writers/jamf_writer.py
+++ b/src/components/policy/tools/template_writers/writers/jamf_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2020 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -186,7 +186,8 @@ class JamfWriter(template_writer.TemplateWriter):
       return
     if 'id' in obj:
       ids_in_ancestry.add(obj['id'])
-    for key, value in obj.items():
+    # Make a copy of items since we are going to change |obj|.
+    for key, value in list(obj.items()):
       if type(value) is not dict:
         continue
       if '$ref' in value:
diff --git a/src/components/policy/tools/template_writers/writers/jamf_writer_unittest.py b/src/components/policy/tools/template_writers/writers/jamf_writer_unittest.py
index 1ab69c41f7..35fc22bdc7 100755
--- a/src/components/policy/tools/template_writers/writers/jamf_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/jamf_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2020 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/json_writer.py b/src/components/policy/tools/template_writers/writers/json_writer.py
index 594362152c..7d5d166c7c 100755
--- a/src/components/policy/tools/template_writers/writers/json_writer.py
+++ b/src/components/policy/tools/template_writers/writers/json_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/json_writer_unittest.py b/src/components/policy/tools/template_writers/writers/json_writer_unittest.py
index 3dacf254cd..5f7c3ad966 100755
--- a/src/components/policy/tools/template_writers/writers/json_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/json_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/mock_writer.py b/src/components/policy/tools/template_writers/writers/mock_writer.py
index bcb43ada06..19e1d74890 100755
--- a/src/components/policy/tools/template_writers/writers/mock_writer.py
+++ b/src/components/policy/tools/template_writers/writers/mock_writer.py
@@ -1,9 +1,9 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-from template_writer import TemplateWriter
+from .template_writer import TemplateWriter
 
 
 class MockWriter(TemplateWriter):
diff --git a/src/components/policy/tools/template_writers/writers/plist_helper.py b/src/components/policy/tools/template_writers/writers/plist_helper.py
index 2f6d3d7750..e69f00dcb6 100755
--- a/src/components/policy/tools/template_writers/writers/plist_helper.py
+++ b/src/components/policy/tools/template_writers/writers/plist_helper.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/plist_strings_writer.py b/src/components/policy/tools/template_writers/writers/plist_strings_writer.py
index cb4d8482fb..adfbc6d279 100755
--- a/src/components/policy/tools/template_writers/writers/plist_strings_writer.py
+++ b/src/components/policy/tools/template_writers/writers/plist_strings_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/plist_writer.py b/src/components/policy/tools/template_writers/writers/plist_writer.py
index acac7e2473..e866648645 100755
--- a/src/components/policy/tools/template_writers/writers/plist_writer.py
+++ b/src/components/policy/tools/template_writers/writers/plist_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/plist_writer_unittest.py b/src/components/policy/tools/template_writers/writers/plist_writer_unittest.py
index 05f7f6c90b..92d75ee7f8 100755
--- a/src/components/policy/tools/template_writers/writers/plist_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/plist_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/reg_writer.py b/src/components/policy/tools/template_writers/writers/reg_writer.py
index 2e279786d0..3fbd0a6b35 100755
--- a/src/components/policy/tools/template_writers/writers/reg_writer.py
+++ b/src/components/policy/tools/template_writers/writers/reg_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/reg_writer_unittest.py b/src/components/policy/tools/template_writers/writers/reg_writer_unittest.py
index e59777f1bc..7278f76b00 100755
--- a/src/components/policy/tools/template_writers/writers/reg_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/reg_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/template_writer.py b/src/components/policy/tools/template_writers/writers/template_writer.py
index 0ce8ed2c64..bc1c3489f7 100755
--- a/src/components/policy/tools/template_writers/writers/template_writer.py
+++ b/src/components/policy/tools/template_writers/writers/template_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -292,9 +292,9 @@ class TemplateWriter(object):
     '''
     for policy in policy_list:
       if policy['type'] == 'group':
-        child_policies = self._GetPoliciesForWriter(policy)
-        child_recommended_policies = filter(self.CanBeRecommended,
-                                            child_policies)
+        child_policies = list(self._GetPoliciesForWriter(policy))
+        child_recommended_policies = list(
+            filter(self.CanBeRecommended, child_policies))
         if child_policies:
           # Only write nonempty groups.
           self.BeginPolicyGroup(policy)
diff --git a/src/components/policy/tools/template_writers/writers/template_writer_unittest.py b/src/components/policy/tools/template_writers/writers/template_writer_unittest.py
index 3d4988e208..d725cdfac7 100755
--- a/src/components/policy/tools/template_writers/writers/template_writer_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/template_writer_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/writer_unittest_common.py b/src/components/policy/tools/template_writers/writers/writer_unittest_common.py
index b0fdf67c48..d96bb7986d 100755
--- a/src/components/policy/tools/template_writers/writers/writer_unittest_common.py
+++ b/src/components/policy/tools/template_writers/writers/writer_unittest_common.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2017 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/xml_formatted_writer.py b/src/components/policy/tools/template_writers/writers/xml_formatted_writer.py
index e4813d6e8b..02fee8293f 100755
--- a/src/components/policy/tools/template_writers/writers/xml_formatted_writer.py
+++ b/src/components/policy/tools/template_writers/writers/xml_formatted_writer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/policy/tools/template_writers/writers/xml_writer_base_unittest.py b/src/components/policy/tools/template_writers/writers/xml_writer_base_unittest.py
index 2b5df21760..40c2ff52bb 100755
--- a/src/components/policy/tools/template_writers/writers/xml_writer_base_unittest.py
+++ b/src/components/policy/tools/template_writers/writers/xml_writer_base_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
